packages/devtools_app/lib/src/memory/memory_timeline.dart

```diff
@@ -121,25 +126,35 @@ class MemoryTimeline {
     postEventSample(EventSample.snapshotEvent(
       DateTime.now().millisecondsSinceEpoch,
       snapshotAuto: auto,
+      events: extensionEvents,
     ));
   }
 
   void addGCEvent() {
     final timestamp = DateTime.now().millisecondsSinceEpoch;
-    postEventSample(EventSample.gcEvent(timestamp));
+    postEventSample(EventSample.gcEvent(
+      timestamp,
+      events: extensionEvents,
```

Comment:
why are we batching up extension events and only posting them when there is another event? it would seem we would want to add each as it occurs with the correct timestamp in stead of pretending they occur at the nearest gc or other event?

Reply:
Because the x-axis is driven by the the ticks returned from the VM.  if we start synthesizing other timestamps then the x-axis will appear stretched as extension events arrive. Possibly we could plot x-axis based on a fix interval and layer the events but then I suspect we'd have the same issue some events would have to be aggregated.